### PR TITLE
Lesson 7. Fruits in general

### PR DESCRIPTION
**Pull request by snoozee**

### CODE REVIEW

File: src/main/java/generics/Box.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        float totalWeight = 0.0F;
+
+        for (T fruit: fruits) {
+            totalWeight += fruit.getSingleWeight();
+        }
+
+        return totalWeight;
+    }
+
+    public boolean compare(Box<?> otherBox) {
```
Comment (SerhiiPasko) [inline src/main/java/generics/Box.java]: `Box<? extends Fruit>` it nothing change but little more clear

File: src/main/java/generics/Main.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        List<Orange> orangeList = new ArrayList<>();
+        for (int i = 0; i< quantity; i++) {
+            orangeList.add(new Orange(1.0F));
+        }
+
+        return orangeList;
+    }
+
+    public static void main(String args[]) {
+        //String[] myArray = {"SBU", "GUR", "ZSU", "TerOborona"};
```
Comment (SerhiiPasko) [inline src/main/java/generics/Main.java]: Please clean-up

File: src/main/java/generics/Main.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
+package generics;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class Main {
+    public static <T> List<T> toList(T array[]) {
```
Comment (SerhiiPasko) [inline src/main/java/generics/Main.java:8]: Already exists Arrays.toList

File: src/main/java/generics/Box.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        return Math.abs(this.getWeight() - otherBox.getWeight()) < 0.0001;
```
Comment (SerhiiPasko) [inline src/main/java/generics/Box.java:44]: All box-types already have compare method. `Float.compare` in our case

**Review by SerhiiPasko — APPROVED** [review]
Cool

File: src/main/java/generics/Main.java (hunk 2 above)
Comment (snoozee) [reply src/main/java/generics/Main.java]: Done

**Pull request merged**